[WebDriver] Support proper WTF logging features

lauromoura opened the pull request:
#### 42ffeb6854c8e0d16e3536a8c87ea5ba40fef290
<pre>
[WebDriver] Support proper WTF logging features
<a href="https://bugs.webkit.org/show_bug.cgi?id=284824">https://bugs.webkit.org/show_bug.cgi?id=284824</a>

Reviewed by Adrian Perez de Castro.

Add the required pumbling to enable WTF LogChannels for
Source/WebDriver. This will allow us add more logging points without
polluting stdout/stderr.

Initially, it&apos;ll have 3 channels: WebDriverClassic, WebDriverBiDi,
SessionHost.

* Source/WebDriver/CMakeLists.txt:
* Source/WebDriver/LogInitialization.cpp: Added.
(WebDriver::logChannels):
(WebDriver::getLogChannel):
* Source/WebDriver/LogInitialization.h: Added.
* Source/WebDriver/Logging.cpp: Added.
* Source/WebDriver/Logging.h: Added.
* Source/WebDriver/PlatformGTK.cmake:
* Source/WebDriver/PlatformPlayStation.cmake:
* Source/WebDriver/PlatformWPE.cmake:
* Source/WebDriver/WebDriverMain.cpp:
(main):
* Source/WebDriver/WebDriverService.cpp:
(WebDriver::WebDriverService::run):
(WebDriver::WebDriverService::acceptHandshake):
(WebDriver::WebDriverService::handleMessage):
(WebDriver::WebDriverService::createSession):
* Source/WebDriver/soup/HTTPServerSoup.cpp:
(WebDriver::HTTPServer::listen):
* Source/WebDriver/soup/WebSocketServerSoup.cpp:
(WebDriver::handleIncomingHandshake):
(WebDriver::handleWebSocketMessage):
(WebDriver::WebSocketServer::listen):
* Source/WebDriver/unix/LoggingUnix.cpp: Added.
(WebDriver::logLevelString):
* Source/WebDriver/win/LoggingWin.cpp: Added.
(WebDriver::logLevelString):
* Source/WebKit/glib/environment-variables.md.in:

Canonical link: <a href="https://commits.webkit.org/288150@main">https://commits.webkit.org/288150@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/140295a1956ef44347f8cb7b3fc30723292aad18

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81790 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1316 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35744 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86337 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32786 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1349 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9136 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63795 "Found 1 new test failure: compositing/patterns/direct-pattern-compositing-position.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21519 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84860 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/980 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74428 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44081 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/880 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28602 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31239 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72262 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29206 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87773 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9030 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6408 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72141 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9215 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70248 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71373 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17831 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15462 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14380 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/402 "") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8981 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14513 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8822 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12345 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10630 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebDriver/LogInitialization.cpp:
```diff
@@ -0,0 +1,74 @@
+
+WTFLogChannel* getLogChannel(const String& name)
+{
+    UNUSED_PARAM(name);
```
You could omit the parameter name instead of using `UNUSED_PARAM()`:

```suggestion
WTFLogChannel* getLogChannel(const String&)
{
```

aperezdc reviewed (commented):
Would it be possible to add a short text in `Source/WebKit/glib/environment-variables.md.in` about the `WEBKIT_DEBUG` environment variable, and list the logging channels added by this patch? I know we currently don't have anything there on this topic, but we have to start somewhere (like [this GTK documentation page](https://docs.gtk.org/gtk4/running.html#gtk_debug), for example).

aperezdc reviewed (approved):
Thanks for updating the patch, LGTM.

lauromoura committed:
[WebDriver] Support proper WTF logging features
https://bugs.webkit.org/show_bug.cgi?id=284824

Reviewed by Adrian Perez de Castro.

Add the required pumbling to enable WTF LogChannels for
Source/WebDriver. This will allow us add more logging points without
polluting stdout/stderr.

Initially, it'll have 3 channels: WebDriverClassic, WebDriverBiDi,
SessionHost.

* Source/WebDriver/CMakeLists.txt:
* Source/WebDriver/LogInitialization.cpp: Added.
(WebDriver::logChannels):
(WebDriver::getLogChannel):
* Source/WebDriver/LogInitialization.h: Added.
* Source/WebDriver/Logging.cpp: Added.
* Source/WebDriver/Logging.h: Added.
* Source/WebDriver/PlatformGTK.cmake:
* Source/WebDriver/PlatformPlayStation.cmake:
* Source/WebDriver/PlatformWPE.cmake:
* Source/WebDriver/WebDriverMain.cpp:
(main):
* Source/WebDriver/WebDriverService.cpp:
(WebDriver::WebDriverService::run):
(WebDriver::WebDriverService::acceptHandshake):
(WebDriver::WebDriverService::handleMessage):
(WebDriver::WebDriverService::createSession):
* Source/WebDriver/soup/HTTPServerSoup.cpp:
(WebDriver::HTTPServer::listen):
* Source/WebDriver/soup/WebSocketServerSoup.cpp:
(WebDriver::handleIncomingHandshake):
(WebDriver::handleWebSocketMessage):
(WebDriver::WebSocketServer::listen):
* Source/WebDriver/unix/LoggingUnix.cpp: Added.
(WebDriver::logLevelString):
* Source/WebDriver/win/LoggingWin.cpp: Added.
(WebDriver::logLevelString):
* Source/WebKit/glib/environment-variables.md.in:

Canonical link: https://commits.webkit.org/288150@main
